release-calendar/src/client/components/TooltipHook.tsx

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {EventApi, View} from '@fullcalendar/core';
+import {EventCard} from './EventCard';
+import {usePopper} from 'react-popper';
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+const TOOLTIP_HEIGHT = 300;
+
+export const Tooltip = (arg: {
+  isMirror: boolean;
+  isStart: boolean;
+  isEnd: boolean;
+  event: EventApi;
+  el: HTMLElement;
+  view: View;
+}): void => {
+  const Content = (): JSX.Element => {
+    return <Hook event={arg.event}></Hook>;
+  };
+
+  ReactDOM.render(<Content />, arg.el);
+};
+
+export const Hook = (children: {event: EventApi}): JSX.Element => {
+  const [referenceElement, setReferenceElement] = React.useState(null);
+  const [popperElement, setPopperElement] = React.useState(null);
+  const [isClicked, setClick] = React.useState(false);
+  const node = React.useRef(null);
+  const {styles, attributes} = usePopper(referenceElement, popperElement, {
+    placement: 'left-end',
+    strategy: 'absolute',
+    modifiers: [
+      {
+        name: 'offset',
+        options: {
+          offset: [-(TOOLTIP_HEIGHT / 2), 5],
+        },
+      },
+      {
+        name: 'flip',
+        options: {
+          flipVariations: false,
+        },
+      },
+      {
+        name: 'eventListeners',
+        options: {
+          scroll: false,
+        },
+      },
+    ],
+  });
+
+  const handleClickOutside = (event: MouseEvent): void => {
+    if (node.current && !node.current.contains(event.target)) {
+      setClick(false);
+    }
+  };
+
+  React.useEffect(() => {
+    document.addEventListener('mousedown', handleClickOutside);
+    return (): void => {
+      document.removeEventListener('mousedown', handleClickOutside);
+    };
+  }, []);
+
+  return (
+    <React.Fragment>
+      <button
+        className={'event-button'}
+        ref={setReferenceElement}
+        onClick={(): void => setClick(!isClicked)}>
+        {children.event.title}
+      </button>
+      {ReactDOM.createPortal(
+        <div ref={node}>
+          <div
+            ref={setPopperElement}
+            style={{...{zIndex: 999}, ...styles.popper}}
```

Comment:
Can this be moved to an SCSS file? Not sure what `popper` suggests

Reply:
There's a couple of things I can do. 
1. Change the CSSProperty in styles.popper directly. 
`styles.popper['zIndex'] = 999;`
2. Keep what I have above
`style={{...{zIndex: 999}, ...styles.popper}}`
3. Or declare `zIndex: 999` in a different variable
`const bringForward: React.CSSProperties = {zIndex: 999};`
`style={{...bringForward,...styles.popper}}`

Popper doesn't have a suggestion for smaller changes to style like this. But when doing a full customized tooltip, they suggest the third (when I wouldn't use styles.popper at all).

Reply:
Hmm, okay. We can leave this as is.